dso/model.py

```diff
@@ -518,14 +535,23 @@ def from_dict(raw: dict):
                     datetime.datetime: dateutil.parser.isoparse,
                     datetime.date: lambda date: datetime.datetime.fromisoformat(date).date(),
                 },
-                cast=[
-                    ArtefactKind,
-                    ComplianceSnapshotStatuses,
-                    MetaRescoringRules,
-                ],
+                cast=[enum.Enum],
             ),
         )
 
+    @property
+    def key(self) -> str:
+        if dataclasses.is_dataclass(self.data):
+            data_key = self.data.key if hasattr(self.data, 'key') else None
+        else:
+            data_key = self.data.get('key')
+
+        return _as_key(self.artefact.key, self.meta.datasource, self.meta.type, data_key)
+
+    @property
+    def id(self) -> str:
+        return hashlib.sha1(self.key.encode('utf-8'), usedforsecurity=False).hexdigest()
+
```

Comment:
as discussed recently in a different context: how about using a shorted digest + how about base64-encoding it (as opposed to hex-str) so save some more chars?

regardless: I think we should safeguard ourselves from accidentally changing this implementation via a unittest (with a comment explaining the implications of changing id-calculation)